src/ERC1155D.sol

```diff
@@ -144,17 +147,9 @@ contract ERC1155D is ERC165, IERC1155, IERC1155MetadataURI {
         address to,
         uint256 id,
         uint256 amount,
-        bytes memory data
+        bytes calldata data
     ) internal virtual {
-        require(to != address(0), "ERC1155: transfer to the zero address");
```

Comment:
If we allow transfers to the zero address I think ditto will break. We use zero address checks to see if the clone exist and assume if the owner is zero the clone has either not been minted or has been burned. If a clone is transferred to the zero address it could be read as not existing in duplicate, but none of the burning logic will have been calculated on it.

Reply:
nice catch. fixed

Reply:
sorry, I force pushed it, so won't be easy to review the fix, but I just added back these lines.